.github/workflows/ci-tests.yaml

```diff
@@ -221,6 +221,32 @@ jobs:
                   path: tests/testing-results
                   retention-days: 30
 
+    gpu-tests:
+        runs-on: [self-hosted, linux, x64]
+        container: ghcr.io/gem5/gcn-gpu:latest
+        timeout-minutes: 180
+        needs: [pre-commit, check-for-change-id]
+        steps:
+            - uses: actions/checkout@v4
+
+            # Build the VEGA_X86/gem5.opt binary.
+            - name: Build VEGA_X86/gem5.opt
+              run: scons build/VEGA_X86/gem5.opt -j`nproc`
+
+            # Run the GPU tests.
```

Comment:
How do I understand what tests this runs?  It does not seem to correspond to any of the other file names in this commit ...

Reply:
The code that follows comment this calls our TestLib framework:

```shell
./main.py run  --skip-build -vvv -t $(nproc) --host gcn_gpu gem5/gpu
```

The important part here is the "gem5/gpu" at the end. That means it's running the tests defined in the "tests/gem5/gpu" directory. For example: https://github.com/gem5/gem5/blob/fe8b2b24eb9b08c07badd6f08fdeedc32db5a5c1/tests/gem5/gpu/test_gpu_apu_se.py. This script show what script gem5 is running and with what parameters using the Vega binary we built in the previous step.

The only snag with this implementation is needing to pass `--host gcn_gpu` to testlib to run the tests. This is because we need to compile VEGA and run the tests in the special docker container and I use this flag to ensure this happens.